### PR TITLE
Feature/ui oldtonew config json hydrator

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Modifies config json's of all plugins to new format
- Creates a unique config json file for each type of plugin (```Table-batchsource.json, Table-batchsink.json```)
- Moves modified config json to cdap-app-template folder
- Modifies plugin-config-factory to fetch configurations from backend instead of node proxy.
- Modifies predefined app templates to new format. [Script to automate the change](https://gist.github.com/ajainarayanan/4d8ce7788697901faa74)
-Modifies the structure of the json that is being passed around directives, services in hydrator 
The default structure of a node across anywhere in hydrator is,
```
{
  name: <Unique-ID-for-name>,
  plugin: {
    name: <Name-of-plugin>,
    label: <Label-from-user>,
    artifact: {...}
    properties: {...}
  },
  _uiPosition: {},
  _backendProperties: {},
  type: <type-of-plugin>
}
```
While constructing the json to be sent to backend only essential properties are retained and the rest will be cleaned up.

newjson

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/widgets/Jms.json [hunk 1]
```diff
@@ -0,0 +1 @@
+undefined
```
Comment (elfenheart) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/widgets/Jms.json]: undefined?

Comment (ajainarayanan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/widgets/Jms.json]: Yeah. Have to work on this. Don't what was the previous JSON. Will fix it.

**Comment by elfenheart** [comment]
:+1: LGTM

File: cdap-app-templates/cdap-etl/cdap-etl-lib/widgets/AmazonSQS.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  "metadata": {
+    "spec-version": "1.0",
+    "artifact-version": "3.3.0-SNAPSHOT",
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/widgets/AmazonSQS.json]: artifact version, plugin name, and plugin type should not be in the metadata. 

Artifact version is redundant, you will be getting properties from a specific artifact already, which requires that you know the version

Type and name don't make sense here, as you will need to know the type and name already in order to look this up. Instead, the property name (filename) should contain the plugin type and name. Plugins with the same name but different types can and do have different config settings.

Comment (nitinmotgi) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/widgets/AmazonSQS.json]: Also, @ajainarayanan please modify the specification document if you make this change. Also move the specification document to wiki.cask.co.

**Comment by albertshau** [comment]
lgtm

**Comment by elfenheart** [comment]
lgtm

**Pull request merged**